pkg/controller/clustersync/clustersync_controller.go

```diff
@@ -540,7 +541,11 @@ func (r *ReconcileClusterSync) applySyncSets(
 				// For non-selector SyncSets we have a more accurate startTime, either ClusterDeployment installedTimestamp
 				// or SyncSet creationTimestamp.
 				logger.WithField("applyTime", applyTime).Debug("observed first successful apply of SyncSet for cluster")
-				metricTimeToApplySyncSet.Observe(applyTime)
+				if syncSetGroup, ok := syncSet.AsMetaObject().GetAnnotations()[constants.SyncSetMetricsGroupAnnotation]; ok {
+					metricTimeToApplySyncSet.WithLabelValues(syncSetGroup).Observe(applyTime)
+				} else {
+					metricTimeToApplySyncSet.WithLabelValues("non-grouped").Observe(applyTime)
```

Comment:
If labels are used in graph index does ''non-grouped' sound better than "none"? I am not sure here.